File: rentCarAngular2/src/app/carDetails.ts
export class CarDetails {
    carDetails_id?: number;
    color?: string;
    weight?: number;
    high?: number;

    constructor(cd: any) {
        this.carDetails_id = cd.carDetails_id;
        this.color = cd.color;
        this.weight = cd.weight;
        this.high = cd.high;
    }
}

File: rentcar/src/main/java/com/przemo/rentcar/entities/cars/CarDetails.java
package com.przemo.rentcar.entities.cars;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class CarDetails
{

    @Id
    private Long carDetails_id;

    @OneToOne(cascade = CascadeType.ALL)
    @MapsId
    @JoinColumn(name = "carId")
    @JsonBackReference
    private Car car;

    private String color;

    private int weight;

    private int high;



}

